tests/common/helpers/voq_lag.py

```diff
@@ -0,0 +1,341 @@
+import time
+import pytest
+import logging
+import re
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.redis import AsicDbCli, AppDbCli, VoqDbCli
+
+TMP_PC = 'PortChannel999'
+
+
+def get_lag_ids_from_chassis_db(duthosts):
+    """
+    Get lag_ids from CHASSIS_DB
+    cmd = 'redis-dump -H 10.0.5.16 -p 6380 -d 12 -y -k "*SYSTEM_LAG_TABLE*PortChannel0016"'
+      Args:
+          duthosts: The duthost fixture.
+
+      Returns:
+          lag_ids<list>: lag id
+
+    """
+    lag_ids = list()
+    for sup in duthosts.supervisor_nodes:
+        voqdb = VoqDbCli(sup)
+        lag_list = voqdb.get_lag_list()
+        for lag in lag_list:
+            lag_ids.append(voqdb.hget_key_value(lag, "lag_id"))
+
+    logging.info("LAG id's preset in CHASSIS_DB are {}".format(lag_ids))
+    return lag_ids
+
+
+def get_lag_id_from_chassis_db(duthosts):
+    """
+    Get LAG id for a lag form CHASSIS_DB
+    Args:
+        duthosts: The duthost fixture.
+
+    Returns:
+        lag_ids <int>: lag id
+    """
+    for sup in duthosts.supervisor_nodes:
+        voqdb = VoqDbCli(sup)
+        lag_list = voqdb.get_lag_list()
+        for lag in lag_list:
+            if TMP_PC in lag:
+                lag_id = voqdb.hget_key_value(lag, "lag_id")
+                logging.info("LAG id for lag {} is {}".format(TMP_PC, lag_id))
+                return lag_id
+
+        pytest.fail("LAG id for lag {} is not preset in CHASSIS_DB".format(TMP_PC))
+
+
+def add_lag(duthost, asic, portchannel_members=None, portchannel_ip=None,
+            portchannel=TMP_PC, add=True):
+    """
+    Add LAG to an ASIC
+    runs command e.g. 'sudo config portchannel -n asic0 add PortChannel99'
+    Args:
+        duthost<object>: duthost
+        asic<object>: asic object
+        portchannel_members<list> : portchannel members
+        portchannel_ip<str>: portchannel ip
+        portchannel<str> : portchannel
+        add<bool> : True adds portchannel
+    """
+    if add:
+        config_cmd = "config portchannel {}"\
+            .format(asic.cli_ns_option if asic.cli_ns_option else "")
+        duthost.shell("{} add {}".format(config_cmd, portchannel))
+        int_facts = duthost.interface_facts(namespace=asic.namespace)['ansible_facts']
+        pytest_assert(int_facts['ansible_interface_facts'][portchannel])
+
+    if portchannel_members:
+        for member in portchannel_members:
+            duthost.shell("config portchannel {} member add {} {}"
+                          .format(asic.cli_ns_option, portchannel, member))
+
+    if portchannel_ip:
+        duthost.shell("config interface {} ip add {} {}"
+                      .format(asic.cli_ns_option, portchannel, portchannel_ip))
+        int_facts = duthost.interface_facts(namespace=asic.namespace)['ansible_facts']
+        pytest_assert(int_facts['ansible_interface_facts']
+                      [portchannel]['ipv4']['address'] == portchannel_ip.split('/')[0])
+        time.sleep(30)
```

Comment:
There are multiple `time.sleep(30)` in this script. Can you replace the hard sleep with utility like `wait_until`? Then we may be able to steal some seconds of running the test script.